### PR TITLE
update rights input field selector

### PR DESCRIPTION
**Pull request by njaffer**
Present tense short summary (50 characters or less)
The right property does not get saved when a New Work is created or edited. 

Here is the debugger output before the change
byebug
    7:     # set the @files ivar then remove the files attribute so it isn't set by default.
=>  8:     files && attributes.delete(:files)
    9:     self.raw_attributes = attributes.dup
   10:     # Files must be attached before saving in order to persist their relationship to the work
   11:     assign_pid && interpret_visibility && attach_files && super && assign_representative
   12:   end
(byebug) attributes
{"title"=>["Test Work 333"], "creator"=>[], "contributor"=>[], "description"=>[], "tag"=>[], "publisher"=>[], "date_created"=>[], "subject"=>[], "language"=>[], "identifier"=>[], "based_near"=>[], "related_url"=>[], "visibility_during_embargo"=>"restricted", "embargo_release_date"=>"2016-02-23", "visibility_after_embargo"=>"open", "visibility_during_lease"=>"open", "lease_expiration_date"=>"2016-02-23", "visibility_after_lease"=>"restricted", "visibility"=>"restricted", "resource_type"=>[]}

Here is the debugger output after the change
(byebug) attributes
{"title"=>["Test Work 333"], "creator"=>[], "contributor"=>[], "description"=>[], "tag"=>[], "rights"=>["http://creativecommons.org/licenses/by-nc/3.0/us/"], "publisher"=>[], "date_created"=>[], "subject"=>[], "language"=>[], "identifier"=>[], "based_near"=>[], "related_url"=>[], "visibility_during_embargo"=>"restricted", "embargo_release_date"=>"2016-02-23", "visibility_after_embargo"=>"open", "visibility_during_lease"=>"open", "lease_expiration_date"=>"2016-02-23", "visibility_after_lease"=>"restricted", "visibility"=>"restricted", "resource_type"=>[]}

Changes proposed in this pull request:
* override multiple? in WorkForm, set rights to true
* 
* 

@projecthydra/sufia-code-reviewers

### CODE REVIEW

File: app/views/records/edit_fields/_rights.html.erb [hunk 1]
```diff
@@ -1,4 +1,4 @@
-<%= f.input :rights, as: :select_with_modal_help,
```
Comment (cam156) [inline app/views/records/edit_fields/_rights.html.erb]: @njaffer The modal help allows for you to display more information about the licenses: 
screen shot 2016-02-22 at 11 05 25 am

I'm not sure why that is not working, but removing it seems like a bad idea.

Comment (cam156) [inline app/views/records/edit_fields/_rights.html.erb]: You are removing the help ? from the screen by removing that class.
screen shot 2016-02-22 at 11 07 11 am

File: app/views/records/edit_fields/_rights.html.erb [hunk 2]
```diff
@@ -1,4 +1,4 @@
...
+<%= f.input :rights, as: :select,
     collection: RightsService.select_options,
-    input_html: { class: 'form-control', multiple: true } %>
+    input_html: { class: 'form-control' } %>
```
Comment (cam156) [inline app/views/records/edit_fields/_rights.html.erb]: @njaffer Has the rights changed from a multi select value to a single value? There is no indication in sufia 6.0 that this has changed: https://github.com/projecthydra/sufia/blob/6.x-stable/sufia-models/app/models/concerns/sufia/generic_file/metadata.rb#L41 

@mjgiarlo was this a decision that was made to move from multiple rights statements to a singular value?

Comment (mjgiarlo) [inline app/views/records/edit_fields/_rights.html.erb]: Not that I can recall, @cam156. There was a ticket for this, but we deferred it: https://github.com/projecthydra/sufia/issues/1287

**Comment by cam156** [comment]
@njaffer looks much better!  Thanks! :+1:   Anyone else want to chime in?

File: app/forms/sufia/forms/work_form.rb [hunk 3]
```diff
@@ -7,5 +7,17 @@ def rendered_terms
                :visibility_after_embargo, :visibility_during_lease,
                :lease_expiration_date, :visibility_after_lease, :visibility, :thumbnail_id, :representative_id, :ordered_member_ids]
     end
+
+    # override Curation Concerns Work Form, Sufia allows multiple values for rights
+    def self.multiple?(term)
+      case term.to_s
+      when 'rights'
+        true
+      when 'ordered_member_ids'
```
Comment (jcoyne) [inline app/forms/sufia/forms/work_form.rb]: Why is this required?

File: spec/form/work_form_spec.rb [hunk 4]
```diff
@@ -0,0 +1,46 @@
+require 'spec_helper'
+
+describe Sufia::Forms::WorkForm do
+  before do
+    class PirateShip < ActiveFedora::Base
```
Comment (jcoyne) [inline spec/form/work_form_spec.rb]: why do this? You can use GenericWork instead.

File: spec/form/work_form_spec.rb [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+      include CurationConcerns::RequiredMetadata
+      include CurationConcerns::BasicMetadata
+      include CurationConcerns::HasRepresentative
+    end
+
+    class PirateShipForm < described_class
```
Comment (jcoyne) [inline spec/form/work_form_spec.rb]: we don't need this either. you can just use `Sufia::Forms::WorkForm`

File: app/forms/sufia/forms/work_form.rb (hunk 3 above)
Comment (njaffer) [inline app/forms/sufia/forms/work_form.rb]: @jcoyne: I am not sure I understand what are you referring to here, can be more specific

Comment (jcoyne) [inline app/forms/sufia/forms/work_form.rb]: How does this clause solve the issue you reported?  It doesn't seem releated to me. If it's not related, it shouldn't go in this PR.

File: spec/form/work_form_spec.rb [hunk 6]
```diff
@@ -0,0 +1,29 @@
+require 'spec_helper'
+
+describe Sufia::Forms::WorkForm do
```
Comment (jcoyne) [inline spec/form/work_form_spec.rb]: This should be testing `Sufia::Forms::GenericWorkForm` instead.  `WorkForm` is an abstract class.

Comment (njaffer) [inline spec/form/work_form_spec.rb]: ok so this is confusing, the form I am testing is work_form and when I use your suggestion, I get an  error: uninitialized constant Sufia::Forms::GenericWorkForm (NameError)

Comment (jcoyne) [inline spec/form/work_form_spec.rb]: Sorry, I see it's actually called `CurationConcerns::GenericWorkForm`

Comment (njaffer) [inline spec/form/work_form_spec.rb]: ok so in my understanding, I am overriding the multiple? in sufia's  work_form and I should be testing sufia work_form?

Comment (jcoyne) [inline spec/form/work_form_spec.rb]: That's up to you. I don't think there's a hard rule for how to test abstract classes.

File: spec/form/work_form_spec.rb (hunk 5 above)
Comment (jcoyne) [inline spec/form/work_form_spec.rb]: Sorry, I meant to say `CurationConcerns::GenericWorkForm`

**Comment by njaffer** [comment]
The PR can be closed. I tested the latest Sufia master branch, and the rights metadata field is getting saved now.

**Comment by njaffer** [comment]
fixes #1645

**Comment by mjgiarlo** [comment]
Thanks for the PR, @njaffer. I see there are some unresolved questions. I'll highlight those.

File: app/forms/sufia/forms/work_form.rb (hunk 3 above)
Comment (mjgiarlo) [inline app/forms/sufia/forms/work_form.rb]: @njaffer This discussion is unresolved.

File: spec/form/work_form_spec.rb (hunk 6 above)
Comment (mjgiarlo) [inline spec/form/work_form_spec.rb]: This one is unresolved too.

File: spec/form/work_form_spec.rb (hunk 4 above)
Comment (mjgiarlo) [inline spec/form/work_form_spec.rb]: This is unresolved.

File: spec/form/work_form_spec.rb (hunk 5 above)
Comment (mjgiarlo) [inline spec/form/work_form_spec.rb]: This is unresolved.

**Comment by mjgiarlo** [comment]
@njaffer If it'd help to pair up on some of this, let me know. (Or if you'll be at LDCX/HDC next week, maybe that'd be a good place to take this up.)

File: app/forms/sufia/forms/work_form.rb (hunk 3 above)
Comment (njaffer) [inline app/forms/sufia/forms/work_form.rb]: @mjgiarlo: I think this was about the comment line I put previously, which is removed now, but I might be wrong.

**Comment by njaffer** [comment]
@mjgiarlo: I have updated the spec test, still there is one unresolved discussion. I reopened the PR on Friday in reference to discussion with @hectorcorrea on #1645. Sorry I had to leave early on Friday, so couldn't cleanup the PR. I will be there at HDC, for sure if there is more discussion needed on this, we can take it up there.

File: app/forms/sufia/forms/work_form.rb (hunk 3 above)
Comment (hectorcorrea) [inline app/forms/sufia/forms/work_form.rb]: @jcoyne @mjgiarlo the reason this method override is required is because without it the `rights` field is being rejected by the controller. This is the method in CC that Nabeela is overriding: https://github.com/projecthydra-labs/curation_concerns/blob/master/app/forms/curation_concerns/forms/work_form.rb#L44

I suspect the fact that the field is defined as multi-value but the CC `multiple?` implementation marks it as single-value causes a mismatch that prevents the controller from allowing this field during the HTTP POST request.

@njaffer I think writing the override as follow might make it clearer that the intention is to handle a specific issue for the `rights` field:

```
def self.multiple?(term)
  return true if term == :rights
  super
end
```

**Comment by mjgiarlo** [comment]
Great :thought_balloon:, @njaffer!

File: app/forms/sufia/forms/work_form.rb (hunk 3 above)
Comment (njaffer) [inline app/forms/sufia/forms/work_form.rb]: @hectorcorrea: thank you for the detailed note.. will update it!
